wen-restart/src/wen_restart.rs

```diff
@@ -2485,6 +2499,106 @@ mod tests {
         );
     }
 
+    fn start_aggregate_heaviest_fork_thread(
+        test_state: &WenRestartTestInitResult,
+        heaviest_fork_slot: Slot,
+        heaviest_fork_bankhash: Hash,
+        exit: Arc<AtomicBool>,
+        expected_error: Option<WenRestartError>,
+    ) -> std::thread::JoinHandle<()> {
+        let progress = wen_restart_proto::WenRestartProgress {
+            state: RestartState::HeaviestFork.into(),
+            my_heaviest_fork: Some(HeaviestForkRecord {
+                slot: heaviest_fork_slot,
+                bankhash: heaviest_fork_bankhash.to_string(),
+                total_active_stake: 1500,
+                shred_version: SHRED_VERSION as u32,
+                wallclock: 0,
+            }),
+            ..Default::default()
+        };
+        let wen_restart_path = test_state.wen_restart_proto_path.clone();
+        let cluster_info = test_state.cluster_info.clone();
+        let bank_forks = test_state.bank_forks.clone();
+        Builder::new()
+            .name("solana-wen-restart-aggregate-heaviest-fork".to_string())
+            .spawn(move || {
+                let result = aggregate_restart_heaviest_fork(
+                    &wen_restart_path,
+                    80,
+                    cluster_info,
+                    bank_forks,
+                    exit,
+                    &mut progress.clone(),
+                );
+                if let Some(expected_error) = expected_error {
+                    assert_eq!(
+                        result.unwrap_err().downcast::<WenRestartError>().unwrap(),
+                        expected_error
+                    );
+                } else {
+                    assert!(result.is_ok());
+                }
+            })
+            .unwrap()
+    }
+
+    #[test]
+    fn test_aggregate_heaviest_fork_send_gossip_early() {
+        let ledger_path = get_tmp_ledger_path_auto_delete!();
+        let test_state = wen_restart_test_init(&ledger_path);
+        let heaviest_fork_slot = test_state.last_voted_fork_slots[0] + 3;
+        let heaviest_fork_bankhash = Hash::new_unique();
+
+        let mut cursor = solana_gossip::crds::Cursor::default();
+        // clear the heaviest fork queue so we make sure a new HeaviestFork is sent out later.
+        let _ = test_state
+            .cluster_info
+            .get_restart_heaviest_fork(&mut cursor);
+
+        let exit = Arc::new(AtomicBool::new(false));
+        let thread = start_aggregate_heaviest_fork_thread(
+            &test_state,
+            heaviest_fork_slot,
+            heaviest_fork_bankhash,
+            exit.clone(),
+            Some(WenRestartError::Exiting),
+        );
+        // Even when everyone else is only seeing itself, we should send out heaviest fork indicating
+        // we have active stake exceeding supermajority.
+        for keypair in test_state.validator_voting_keypairs.iter().skip(6) {
```

Comment:
Instead of a `.skip` here, makes more sense to to a `.take(supermajority_ratio)`

Edit: There seems to be an inborn assumption that the validator itself is the first key, which is why we're skipping here.

Hidden assumptions like this are best checked via an assert

Reply:
Also best avoid magic numbers like `6`, we should be computing using well known constants like the supermajority stake

Reply:
Done.